src/app/space-template/space-template.service.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { async, inject, TestBed } from '@angular/core/testing';
+import { BaseRequestOptions, Http, Response, ResponseOptions } from '@angular/http';
+import { MockBackend } from '@angular/http/testing';
+
+import { cloneDeep } from 'lodash';
+
+import { AuthenticationService, UserService, AUTH_API_URL, User } from 'ngx-login-client';
+import { Broadcaster, Logger } from 'ngx-base';
+import { WIT_API_URL } from "../api/wit-api";
+import { SpaceTemplateService } from './space-template.service';
+import { SpaceTemplate } from './space-template';
+
+describe('Service: SpaceTemplateService', () => {
+  let mockService: MockBackend;
+  let fakeAuthService: any;
+  let service: SpaceTemplateService;
+
+  beforeEach(() => {
+    fakeAuthService = {
+      getToken: function () {
+        return '';
+      },
+      isLoggedIn: function () {
+        return true;
+      }
+    };
+    TestBed.configureTestingModule({
+      providers: [
+        Logger,
+        BaseRequestOptions,
+        MockBackend,
+        {
+          provide: Http,
+          useFactory: (backend: MockBackend,
+                       options: BaseRequestOptions) => new Http(backend, options),
+          deps: [MockBackend, BaseRequestOptions]
+        },
+        {
+          provide: AuthenticationService,
+          useValue: fakeAuthService
+        },
+        {
+          provide: WIT_API_URL,
+          useValue: "http://example.com"
+        },
+        {
+          provide: AUTH_API_URL,
+          useValue: 'http://example.com/auth'
+        },
+        SpaceTemplateService,
+        Broadcaster
+      ]
+    });
+  });
+
+  beforeEach(inject(
+    [MockBackend, SpaceTemplateService],
+    (mock: MockBackend, serviceInjected: SpaceTemplateService) => {
+      mockService = mock;
+      service = serviceInjected;
+    }
+  ));
+
+let spaceTemplates = [
```

Comment:
This will most likely need to change once we finalize the space template PR in core. The model for link types and link type combinations WILL change.